### ISSUE TITLE
cwr fails if jenkins job already exists

### ISSUE THREAD

**Issue by kwmonroe** [juju-solutions/layer-cwr#7]
I stumbled upon this when upgrading my `cwr` charm.  I happened to remove the jenkins relation first, and then hit this problem when i added it back.

Repro:
- juju deploy cwr-ci
- juju remove-relation cwr jenkins
- juju add-relation cwr jenkins

Result:
```
unit-cwr-1: 20:29:07 INFO unit.cwr/1.juju-log jenkins:7: Invoking reactive handler: hooks/relations/jenkins-extension/requires.py:11:joined
unit-cwr-1: 20:29:07 INFO unit.cwr/1.juju-log jenkins:7: Updating extension interface with up-to-date data.
unit-cwr-1: 20:29:07 INFO unit.cwr/1.juju-log jenkins:7: Invoking reactive handler: reactive/cwr.py:46:install_jenkins_jobs
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed Traceback (most recent call last):
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/charm/hooks/jenkins-relation-changed", line 19, in <module>
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     main()
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/.venv/lib/python3.5/site-packages/charms/reactive/__init__.py", line 78, in main
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     bus.dispatch()
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/.venv/lib/python3.5/site-packages/charms/reactive/bus.py", line 434, in dispatch
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     _invoke(other_handlers)
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/.venv/lib/python3.5/site-packages/charms/reactive/bus.py", line 417, in _invoke
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     handler.invoke()
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/.venv/lib/python3.5/site-packages/charms/reactive/bus.py", line 291, in invoke
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     self._action(*args)
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/charm/reactive/cwr.py", line 60, in install_jenkins_jobs
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     jclient.create_job(subdirname, configxml)
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed   File "/var/lib/juju/agents/unit-cwr-1/.venv/lib/python3.5/site-packages/jenkins/__init__.py", line 987, in create_job
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed     raise JenkinsException('job[%s] already exists' % (name))
unit-cwr-1: 20:29:08 INFO unit.cwr/1.jenkins-relation-changed jenkins.JenkinsException: job[RunCwr] already exists
```

Possible solutions:

- try/except and pass if the exception is 'already exists' here: https://github.com/juju-solutions/layer-cwr/blob/master/reactive/cwr.py#L60
- add support for `departed` in `interface-jenkins-extension`.  maybe it sets a `departing` state that `cwr` can react to:
    ```
    @when('jenkins.jobs.ready', 'jenkins.departing')
    def oh_crap(jenkins):
        // remove jobs
    ```

**Comment by ktsakalozos** [comment]
I would +1 the "oh_crap" approach and try to remove the jobs on jenkins departure.

-1 on the pass on "already exists". We should not proceed installation if the jobs to be deployed are already available. We should not overwrite the  jobs because the user might be doing something custom in the already existing copies and we should not silence this error because this will compromise the docomented/expected behaviour of the cwr charm.

**Comment by kwmonroe** [comment]
closed by #10
